lib/shower/index.js

```diff
@@ -1,3 +1,100 @@
-export { default as Container } from './container';
-export { default as defaultOptions } from './default-options';
-export { default as Shower } from './shower';
+import { EventTarget } from '../utils';
+import defaultOptions from './default-options';
+import Slide from './slide';
+import Player from './player';
+import Plugins from './plugins';
+
+const ensureId = (slideElement, index) => {
+    if (!slideElement.id) {
+        slideElement.id = index + 1;
+    }
+
+    return slideElement;
+};
+
+/**
+ * Core module of the Shower.
+ *
+ * @param {object} [options] Shower options.
+ */
+class Shower extends EventTarget {
+    constructor(options) {
+        super();
+        this.options = Object.freeze(Object.assign({}, defaultOptions, options));
+
+        const containerSelector = this.options.container_selector;
+        this.container = document.querySelector(containerSelector);
+        if (!this.container) {
+            throw new Error(`Shower container with selector '${containerSelector}' not found.`);
+        }
+
+        this._initPlayer();
+        this._initMode();
+        this.plugins = new Plugins(this);
+    }
```

Comment:
This class is hardly doing anything anymore beside initializing player. Its remaining methods seem to belong in the player.